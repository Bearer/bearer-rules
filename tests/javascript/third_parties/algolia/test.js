const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  
  test("datatype_in_index", ({ expect }) => {
    const testCase = "datatype_in_index.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test("datatype_in_save_object", ({ expect }) => {
    const testCase = "datatype_in_save_object.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})