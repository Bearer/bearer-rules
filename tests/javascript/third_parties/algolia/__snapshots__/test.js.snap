// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`javascript_third_parties_algolia datatype_in_index 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "javascript_third_parties_algolia",
      "title": "Leakage of sensitive data to Algolia",
      "description": "## Description\\nLeaking sensitive data to third-party data tools is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Algolia.\\n\\n## Remediations\\nIf you really need to identify users, use unique identifiers from the database.\\n\\n\`\`\`javascript\\n  const algoliaSearch = require('algoliasearch')\\n  const myAlgolia = algoliaSearch(\\"123\\", \\"123\\")\\n  const index = myAlgolia.initIndex(user.uuid)\\n\`\`\`\\n\\n## Resources\\n- [Algolia docs](https://www.algolia.com/doc/)\\n- [OWASP logging cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_third_parties_algolia",
      "line_number": 4,
      "full_filename": "/tmp/bearer-scan/datatype_in_index.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "7a794bd6-a6d1-429d-91a2-377acce9e9db",
        "name": "Bank Account"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 35,
          "end": 62
        }
      },
      "sink": {
        "start": 4,
        "end": 4,
        "column": {
          "start": 15,
          "end": 63
        },
        "content": "myAlgolia.initIndex(company.bank_account_number)"
      },
      "parent_line_number": 4,
      "snippet": "myAlgolia.initIndex(company.bank_account_number)",
      "fingerprint": "3b8a9db66a5d96671524cd36226b34d5_0",
      "old_fingerprint": "06d0cf262781955d2293578d29424a6f_0",
      "code_extract": "const index = myAlgolia.initIndex(company.bank_account_number)"
    }
  ]
}"
`;

exports[`javascript_third_parties_algolia datatype_in_save_object 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "javascript_third_parties_algolia",
      "title": "Leakage of sensitive data to Algolia",
      "description": "## Description\\nLeaking sensitive data to third-party data tools is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Algolia.\\n\\n## Remediations\\nIf you really need to identify users, use unique identifiers from the database.\\n\\n\`\`\`javascript\\n  const algoliaSearch = require('algoliasearch')\\n  const myAlgolia = algoliaSearch(\\"123\\", \\"123\\")\\n  const index = myAlgolia.initIndex(user.uuid)\\n\`\`\`\\n\\n## Resources\\n- [Algolia docs](https://www.algolia.com/doc/)\\n- [OWASP logging cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_third_parties_algolia",
      "line_number": 8,
      "full_filename": "/tmp/bearer-scan/datatype_in_save_object.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "b5a3b0fd-dd5c-420d-91ce-dd2dddc8cc38",
        "name": "IP address"
      },
      "category_groups": [
        "Personal Data"
      ],
      "source": {
        "start": 7,
        "end": 7,
        "column": {
          "start": 19,
          "end": 43
        }
      },
      "sink": {
        "start": 8,
        "end": 9,
        "column": {
          "start": 1,
          "end": 65
        },
        "content": "index\\n  .saveObject(userObj, { autoGenerateObjectIDIfNotExist: true })"
      },
      "parent_line_number": 8,
      "snippet": "index\\n  .saveObject(userObj, { autoGenerateObjectIDIfNotExist: true })",
      "fingerprint": "3b8a9db66a5d96671524cd36226b34d5_0",
      "old_fingerprint": "38d2aaad78cefccc70a267a9100219ff_0",
      "code_extract": "index\\n  .saveObject(userObj, { autoGenerateObjectIDIfNotExist: true })"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "javascript_third_parties_algolia",
      "title": "Leakage of sensitive data to Algolia",
      "description": "## Description\\nLeaking sensitive data to third-party data tools is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Algolia.\\n\\n## Remediations\\nIf you really need to identify users, use unique identifiers from the database.\\n\\n\`\`\`javascript\\n  const algoliaSearch = require('algoliasearch')\\n  const myAlgolia = algoliaSearch(\\"123\\", \\"123\\")\\n  const index = myAlgolia.initIndex(user.uuid)\\n\`\`\`\\n\\n## Resources\\n- [Algolia docs](https://www.algolia.com/doc/)\\n- [OWASP logging cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_third_parties_algolia",
      "line_number": 12,
      "full_filename": "/tmp/bearer-scan/datatype_in_save_object.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 12,
        "end": 12,
        "column": {
          "start": 29,
          "end": 39
        }
      },
      "sink": {
        "start": 12,
        "end": 12,
        "column": {
          "start": 1,
          "end": 43
        },
        "content": "index.saveObjects([{ email: user.email }])"
      },
      "parent_line_number": 12,
      "snippet": "index.saveObjects([{ email: user.email }])",
      "fingerprint": "3b8a9db66a5d96671524cd36226b34d5_1",
      "old_fingerprint": "38d2aaad78cefccc70a267a9100219ff_1",
      "code_extract": "index.saveObjects([{ email: user.email }])"
    }
  ]
}"
`;
